arangod/Cluster/RebootTracker.cpp

```diff
@@ -66,296 +84,135 @@ RebootTracker::RebootTracker(RebootTracker::SchedulerPointer scheduler)
-        } catch (std::exception const& ex) {
-          LOG_TOPIC("3d935", INFO, Logger::CLUSTER)
-              << "Failed to execute reboot callback: " << descr << ": "
-              << ex.what();
-        } catch (...) {
-          LOG_TOPIC("f7427", INFO, Logger::CLUSTER)
-              << "Failed to execute reboot callback: " << descr << ": "
-              << "Unknown error.";
-        }
-      }
-    }
-  };
-}
-
-void RebootTracker::queueCallbacks(
-    std::vector<
-        std::shared_ptr<std::unordered_map<CallbackId, DescriptedCallback>>>
-        callbacks) {
-  if (callbacks.empty()) {
+  auto& reboots = it->second;
+  // we expect if current reboot id smaller than requested
+  // we execute callback immediately
+  TRI_ASSERT(from <= reboots.cbegin()->first);
+  if (to > reboots.crbegin()->first) {
+    queueCallbacks(it);
     return;
   }
-
-  TRI_ASSERT(std::none_of(callbacks.cbegin(), callbacks.cend(),
-                          [](auto it) { return it == nullptr; }));
-  TRI_ASSERT(std::none_of(callbacks.cbegin(), callbacks.cend(),
-                          [](auto it) { return it->empty(); }));
-
-  auto cb = createSchedulerCallback(std::move(callbacks));
-  _scheduler->queue(RequestLane::CLUSTER_INTERNAL, std::move(cb));
+  std::vector<RebootIds::node_type> callbacks;
+  for (auto rbIt = reboots.begin(); rbIt->first < to;) {
+    TRI_ASSERT(rbIt != reboots.end());
+    callbacks.push_back(reboots.extract(rbIt++));
+  }
+  TRI_ASSERT(!reboots.empty());
+  _scheduler->queue(RequestLane::CLUSTER_INTERNAL,
+                    [callbacks = std::move(callbacks)]() mutable noexcept {
+                      for (auto& calls : callbacks) {
+                        for (auto& e : calls.mapped()) {
+                          safeInvoke(e.second);
+                        }
+                      }
+                    });
 }
 
-void RebootTracker::unregisterCallback(PeerState const& peerState,
-                                       RebootTracker::CallbackId callbackId) {
-  MUTEX_LOCKER(guard, _mutex);
+void RebootTracker::queueCallback(DescriptedCallback&& callback) noexcept {
+  _scheduler->queue(RequestLane::CLUSTER_INTERNAL,
+                    [callback = std::move(callback)]() mutable noexcept {
+                      safeInvoke(callback);
+                    });
+}
 
-  auto const cbIt = _callbacks.find(peerState.serverId());
-  if (cbIt != _callbacks.end()) {
-    auto& rebootMap = cbIt->second;
-    auto const rbIt = rebootMap.find(peerState.rebootId());
-    if (rbIt != rebootMap.end()) {
-      auto& callbackSetPtr = rbIt->second;
-      TRI_ASSERT(callbackSetPtr != nullptr);
-      callbackSetPtr->erase(callbackId);
-      if (callbackSetPtr->empty()) {
-        rebootMap.erase(rbIt);
+void RebootTracker::unregisterCallback(std::string_view serverId,
+                                       RebootId rebootId,
+                                       CallbackId callbackId) noexcept {
+  std::lock_guard guard{_mutex};
```

Comment:
There could theoretically be an exception, I ignore this.
As an alternative, I can use `absl::Mutex`, but I don't think it's necessary, because I think that in a properly loaded program in the correct code, `std::mutex` is also don't throw.
`noexcept` is needed here because the code is called from the dtor